src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -1167,8 +1172,7 @@ private List<CompletionResult> GetResultForString(CompletionContext completionCo
                     cursorIndexInString = strValue.Length;
 
                 var analysis = new CompletionAnalysis(_ast, _tokens, _cursorPosition, _options);
-                var subContext = analysis.CreateCompletionContext(completionContext.ExecutionContext);
-                subContext.Helper = completionContext.Helper;
+                var subContext = analysis.CreateCompletionContext(completionContext.TypeInferenceContext.Helper.CurrentPowerShell);
```

Comment:
> completionContext.TypeInferenceContext.Helper.CurrentPowerShell

Why not just `completionContext.Helper.CurrentPowerShell`?

On the second thought, a single `Helper` was shared by `completionContext` and `subContext` previously, now new objects `TypeInferenceContext` and `PowerShellExecutionHelper` will be created. It looks to me `TypeInferenceContext` is stateless and can be shared by two `CompletionContext`, so it seems reasonable to continue shared the same objects.

How about add an overload of `CompletionContext.CreateCompletionContext`? Like this:
```CSharp
internal CompletionContext CreateCompletionContext(PowerShell powerShell)
{
    var typeInferenceContext = new TypeInferenceContext(powerShell)
    {
        CurrentTypeDefinitionAst = Ast.GetAncestorTypeDefinitionAst(asts.Last())
    };
    InitCompletionContext(typeInferenceContext)
}

internal CompletionContext CreateCompletionContext(TypeInferenceContext typeinferenceContext)
{
    InitCompletionContext(typeInferenceContext)
}

internal void InitCompletionContext(TypeInferenceContext typeinferenceContext)
{
    // Existing init code
    ...
    return new CompletionContext
    {
        ...
        ExecutionContext = typeinferenceContext.ExecutionContext,
        TypeInferenceContext = typeInferenceContext,
        Helper = typeInferenceContext.Helper,
        ...
    }
}
```
So `subContext` can be created via `var subContext = analysis.CreateCompletionContext(completionContext.TypeInferenceContext)` and share the same `TypeInferenceContext` and `PowerShellExecutionHelper` with `completionContext`.